### ISSUE TITLE
compose `.in` requirements declaratively rather than imperatively

### ISSUE THREAD

**Issue by cfm** [freedomofpress/securedrop#6931]
## Description

`make update-python3-requirements` composes our layers of `.in` requirements imperatively, e.g.:

https://github.com/freedomofpress/securedrop/blob/960c43b47a4a0a90e261d2206fe4aa9375622e25/Makefile#L32-L37

Composing them declaratively, via `-r` directives in the `.in` files themselves, would make it easier to reason about requirements composition at the specifications themselves, rather than having to refer to the specific `pip-compile` invocations in the Makefile.

## How will this impact [SecureDrop users](https://github.com/freedomofpress/securedrop-ux/wiki/Users)?

No user-facing changes.

## How would this affect SecureDrop's [threat model](https://docs.securedrop.org/en/stable/threat_model/threat_model.html)?

No threat-model implications.

**Comment by rocodes** [comment]
Yup, and already being done in some other repos: [securedrop-client](https://github.com/freedomofpress/securedrop-client/blob/38aa87d13b5b400e7ea1f682a12f90747f47f0fb/requirements/dev-bookworm-requirements.in#L1), [securedrop-export](https://github.com/freedomofpress/securedrop-export/blob/3c10e04e0b11a382d97efb54f11030f64487be08/requirements/dev-bullseye-requirements.in#L2) 

In favour of this along with having more cross project Makefile standardization generally
